homeassistant/components/climate/evohome.py

```diff
@@ -53,156 +65,420 @@
+        """
+        is_off = \
+            self.target_temperature == self.min_temp and \
+            self._status['setpointStatus']['setpointMode'] == EVO_PERMOVER
+        return not is_off
+
+    def _set_temperature(self, temperature, until=None):
+        """Set the new target temperature of a Zone.
+
+        temperature is required, until can be:
+          - strftime('%Y-%m-%dT%H:%M:%SZ') for TemporaryOverride, or
+          - None for PermanentOverride (i.e. indefinitely)
+        """
+        try:
+            self._obj.set_temperature(temperature, until)
+        except HTTPError as err:
+            if not self._handle_exception("HTTPError", str(err)):               # noqa: E501; pylint: disable=no-member
+                raise
+
+    def set_temperature(self, **kwargs):
+        """Set new target temperature, indefinitely."""
+        self._set_temperature(kwargs['temperature'], until=None)
+
+    def turn_on(self):
+        """Turn the evohome Zone on.
+
+        This is achieved by setting the Zone to its 'FollowSchedule' mode.
+        """
+        self._set_operation_mode(EVO_FOLLOW)
+
+    def turn_off(self):
+        """Turn the evohome Zone off.
+
+        This is achieved by setting the Zone to its minimum temperature,
+        indefinitely (i.e. 'PermanentOverride' mode).
+        """
+        self._set_temperature(self.min_temp, until=None)
+
+    def set_operation_mode(self, operation_mode):
+        """Set an operating mode for a Zone.
+
+        Currently limited to 'Auto' & 'Manual'. If 'Off' is needed, it can be
+        enabled via turn_off method.
+
+        NB: evohome Zones do not have an operating mode as understood by HA.
+        Instead they usually 'inherit' an operating mode from their controller.
+
+        More correctly, these Zones are in a follow mode, 'FollowSchedule',
+        where their setpoint temperatures are a function of their schedule, and
+        the Controller's operating_mode, e.g. Economy mode is their scheduled
+        setpoint less (usually) 3C.
+
+        Thus, you cannot set a Zone to Away mode, but the location (i.e. the
+        Controller) is set to Away and each Zones's setpoints are adjusted
+        accordingly to some lower temperature.
+
+        However, Zones can override these setpoints, either for a specified
+        period of time, 'TemporaryOverride', after which they will revert back
+        to 'FollowSchedule' mode, or indefinitely, 'PermanentOverride'.
+        """
+        self._set_operation_mode(HA_STATE_TO_ZONE.get(operation_mode))
+
+    def _set_operation_mode(self, operation_mode):
+        if operation_mode == EVO_FOLLOW:
+            try:
+                self._obj.cancel_temp_override(self._obj)
+            except HTTPError as err:
+                if not self._handle_exception("HTTPError", str(err)):           # noqa: E501; pylint: disable=no-member
+                    raise
```

Comment:
Log error and return.

Reply:
This `raise` shouldn't have been there - it's a regression, and has now been removed.